src/components/views/rooms/BasicMessageComposer.js

```diff
@@ -392,6 +430,42 @@ export default class BasicMessageEditor extends React.Component {
         return caretPosition;
     }
 
+    _wrapSelectionAsInline(prefix, suffix = prefix) {
+        const range = getRangeForSelection(
+            this._editorRef,
+            this.props.model,
+            document.getSelection());
+        formatInline(range, prefix, suffix);
+    }
+
+    _formatBold = () => {
+        this._wrapSelectionAsInline("**");
+    }
+
+    _formatItalic = () => {
+        this._wrapSelectionAsInline("*");
```

Comment:
Should this not be toggling it off when the user already has it italicized? (same for others)

Reply:
Ideally yes, but it's not part of the MVP and would be a bit trickier to do.